app/views/CreditAndRefunds.scala.html

```diff
@@ -111,7 +111,22 @@
         <h2 class="govuk-heading-s">
             <span class="govuk-body">@messages("credit-and-refund.subHeading.has-credits-1") </span>
             @maybeAvailableCredit.toCurrency
-            <span class="govuk-body">@messages(if(isAgent) "credit-and-refund.agent.subHeading.has-credits-2" else "credit-and-refund.subHeading.has-credits-2")</span>
+            <!--Checks if there are any items in the list, depending on that ":" or "." message is used.-->
+            @{
+                val listSize = creditCharges.filter(x => {
```

Comment:
any reason this end up in the view but not in the controller level ??? 
@joesheridan  / @SiddharthAtos

Reply:
will be handled in following ticket. 
https://jira.tools.tax.service.gov.uk/browse/MISUV-4644

the code was added to add Period. Implementing on controller was breaking other on-page entries. There were some missing condition which were not handled